### PR TITLE
Set of uart changes to support all uart test suites on nrf54l15pdk 

### PR DESCRIPTION
**Pull request by nordic-krch**
Cherry-picks from upstream and upstream PRs to support full UART test suite on nrf54l15pdk (and patially on nrf54h20dk).

### CODE REVIEW

**Comment by nordic-segl** [comment]
I see that
drivers: serial: nrf: Adapt config dependencies to nrf54h20
https://github.com/zephyrproject-rtos/zephyr/pull/71365
was merged in the upstream Zephyr.

**Review by nordic-segl — APPROVED** [review]

File: tests/drivers/uart/uart_elementary/README.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+The purpose of this test is to validate basic UART driver functions,
+that are not tested elsewhere.
+UART interrupt mode is used for the tests purpose.
+
+Hardware setup required for these tests:
+For single uart conviguration - UART0 TX connected to RX and CTS to RTS
```
Comment (nika-nordic) [inline tests/drivers/uart/uart_elementary/README.txt:6]: typo - `configuration`

File: tests/drivers/uart/uart_elementary/README.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+For dual uart configuratiom - UART0 and UART1 TXs and RXs cross-connected
```
Comment (nika-nordic) [inline tests/drivers/uart/uart_elementary/README.txt:7]: typo - `configuration`

**Review by nika-nordic — APPROVED** [review]

File: tests/drivers/uart/uart_elementary/README.txt (hunk 2 above)
Comment (nordic-krch) [reply tests/drivers/uart/uart_elementary/README.txt:7]: i think we can leave with it and get it fixed in upstream.

File: tests/drivers/uart/uart_elementary/README.txt (hunk 1 above)
Comment (nordic-krch) [reply tests/drivers/uart/uart_elementary/README.txt:6]: i think we can leave with it and get it fixed in upstream.

**Pull request merged**